src/dipdup/index.py

```diff
@@ -405,13 +415,19 @@ async def _on_match(
 
         if matched_big_map.action.has_key:
             key_type = handler_config.key_type_cls
-            key = key_type.parse_obj(matched_big_map.key)
+            try:
+                key = key_type.parse_obj(matched_big_map.key)
+            except ValidationError:
```

Comment:
`except Error as e raise Error from e` as first exception is a direct cause of second one.